s4e-backend/src/main/java/pl/cyfronet/s4e/bean/Invitation.java

```diff
@@ -0,0 +1,41 @@
+package pl.cyfronet.s4e.bean;
+
+import lombok.*;
+import pl.cyfronet.s4e.bean.audit.CreationAndModificationAudited;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+
+@Entity
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "invitation", uniqueConstraints = @UniqueConstraint(columnNames = {"email", "institution_id"}))
+public class Invitation extends CreationAndModificationAudited {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @EqualsAndHashCode.Include
+    private Long id;
+
+    @NotEmpty
+    @EqualsAndHashCode.Include
+    private String email;
+
+    @NotEmpty
+    @Column(unique = true)
+    @EqualsAndHashCode.Include
+    private String token;
+
+    @NotNull
+    @EqualsAndHashCode.Include
+    private LocalDateTime expirationTimestamp;
+
+    @ManyToMany(fetch = FetchType.LAZY)
```

Comment:
This should be a many-to-one association with an additional `optional = false`. The fetch mode though is good.